modules/nextflow/src/main/groovy/nextflow/k8s/client/K8sClient.groovy

```diff
@@ -244,20 +244,22 @@ class K8sClient {
             return state
         }
 
-        if( status?.phase == 'Pending' && status.conditions instanceof List ) {
-            final allConditions = status.conditions as List<Map>
-            final cond = allConditions.find { cond -> cond.type == 'PodScheduled' }
-            if( cond.reason == 'Unschedulable' ) {
-                def message = "K8s pod cannot be scheduled"
-                if( cond.message ) message += " -- $cond.message"
-                //def cause = new K8sResponseException(resp)
-                log.warn1(message)
+        if( status?.phase == 'Pending' ){
+            if( status.conditions instanceof List ) {
+                final allConditions = status.conditions as List<Map>
+                final cond = allConditions.find { cond -> cond.type == 'PodScheduled' }
+                if( cond.reason == 'Unschedulable' ) {
+                    def message = "K8s pod cannot be scheduled"
+                    if( cond.message ) message += " -- $cond.message"
+                    //def cause = new K8sResponseException(resp)
+                    log.warn1(message)
+                }
             }
             // undetermined status -- return an empty response
             return Collections.emptyMap()
         }
 
-        throw new K8sResponseException("K8s invalid pod status (missing container status)", resp)
+        throw new K8sResponseException("K8s Can't determine pod state from status returned by K8s API ", resp)
     }
```

Comment:
Frankly, I think the previous err message was a bit more informative.

Reply:
@pditommaso The previous message did not reflect why the Exception was being thrown anymore.  The problem is not missing container status, but rather that K8s returned a set of phase and conditions that the code can not determine the pod state (end of function without any resolvement).

I can revert the change to the message, or insert a more informative message of your choice :) 


_I'm really bad at constructing helpful error message btw :)_

Reply:
Fair enough, what about `K8s undermined status conditions for pod $podName` ?

Reply:
> Fair enough, what about `K8s undermined status conditions for pod $podName` ?

Sounds good to me.  Will update pull request.